File: public/js/game/commands/connectToRoomCommand.js
function ConnectToRoomCommand(roomId){
    Command.call(this, ConnectToRoomCommand.ID);
    this.roomId = roomId
    this.data = true;
    this.execute = function(){
        Model.room = this.data.room;
        Model.dispatch(Events.ROOM_READY)
        Model.dispatch(Events.ROOM_USERS_CHANGE)
        trace("Room ready");
    }
}
ConnectToRoomCommand.ID = 'roomconn'
Command.add(ConnectToRoomCommand.ID, ConnectToRoomCommand)

File: src/main/java/com/hellsepontus/commands/ConnectToRoomCommand.java
package com.hellsepontus.commands;

import com.hellsepontus.model.Room;

import java.util.Collections;

public class ConnectToRoomCommand extends Command {
    
    public static final String ID = "roomconn";
    public int roomId;
    
    public ConnectToRoomCommand() {
        super(ID);
    }

    @Override
    public void execute() {
        Room room = manager.model().getOrCreateRoom(this.roomId);
        if(room.containsUser(getUser().uid))
            throw new RuntimeException("Already in room");
        room.addUser(getUser());
        send(ID, Collections.singletonMap("room", (Object) room));
        new AddRoomUserCommand(room, getUser()).execute();
    }
}
